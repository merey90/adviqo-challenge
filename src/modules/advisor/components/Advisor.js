import React from 'react';
import {
  TableCell,
  TableRow,
  Tooltip
} from '@material-ui/core';
import { withStyles } from '@material-ui/core/styles';
import RadioButtonChecked from '@material-ui/icons/RadioButtonChecked';
import RadioButtonUnchecked from '@material-ui/icons/RadioButtonUnchecked';

const styles = theme => ({
  tableCell: {
    paddingLeft: '5px',
    paddingRight: '4px'
  }
});

function Advisor(props) {
  const { advisor, classes } = props;
  return (
    <TableRow>
      <TableCell className={classes.tableCell}>{advisor.fullName}</TableCell>
      <TableCell className={classes.tableCell}>{advisor.language}</TableCell>
      <TableCell className={classes.tableCell}>
      <Tooltip
        title={!!advisor.status ? 'online':'offline'}
        enterDelay={300}
      >
        {!!advisor.status ?
          <RadioButtonChecked
            fontSize="large"
            color="primary"
          /> :
          <RadioButtonUnchecked
            fontSize="large"
            color="error"
          />
        }
      </Tooltip>
      </TableCell>
      <TableCell className={classes.tableCell} align={'right'}>
        {advisor.reviews}
      </TableCell>
    </TableRow>
  );
}

export default withStyles(styles)(Advisor);