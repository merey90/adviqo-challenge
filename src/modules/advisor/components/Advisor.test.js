import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import Advisor from './Advisor';

const advisor = {
  id: 1,
  fullName: 'Merey Zholdas',
  language: 'Kazakh',
  status: 'online',
  reviews: 4
};

it('renders without crashing', () => {
  shallow(<Advisor key={advisor.id} advisor={advisor} />);
});

test('render Advisor with proper values', () => {
  const component = renderer.create(
    <Advisor key={advisor.id} advisor={advisor} />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});