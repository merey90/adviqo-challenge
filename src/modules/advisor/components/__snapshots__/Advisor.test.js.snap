// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Advisor 1`] = `
<Advisor
  advisor={
    Object {
      "fullName": "Merey Zholdas",
      "id": 1,
      "language": "Kazakh",
      "reviews": 4,
      "status": "online",
    }
  }
  classes={
    Object {
      "tableCell": "Advisor-tableCell-1",
    }
  }
/>
`;
